diagnostic_feedback/data_tool.py

```diff
@@ -0,0 +1,161 @@
+            return dict(report_store.links_for(course_key)).get(self.last_export_result['report_filename'])
+        except Exception:
+            pass
+
+    def _get_status(self):
+        self.check_pending_export()
+        log.info("------------ in _get_status - return status ---------------")
+        log.info(self.download_url_for_last_report)
+        return {
+            'export_pending': bool(self.active_export_task_id),
+            'last_export_result': self.last_export_result,
+            'download_url': self.download_url_for_last_report,
+        }
+
+    def check_pending_export(self):
+        """
+        If we're waiting for an export, see if it has finished, and if so, get the result.
+        """
+        from .tasks import export_data as export_data_task  # Import here since this is edX LMS specific
+        if self.active_export_task_id:
+            log.info("------------ in check_pending_export - checking status ---------------")
+            async_result = export_data_task.AsyncResult(self.active_export_task_id)
+            if async_result.ready():
+                self._save_result(async_result)
+
+    @XBlock.json_handler
+    def start_export(self, data, suffix=''):
+        """ Start a new asynchronous export """
+        log.info("------------ in start_export ---------------")
+        root_block_id = self.scope_ids.usage_id
+        root_block_id = unicode(getattr(root_block_id, 'block_id', root_block_id))
+        #
+
+        if not self.user_is_staff():
+            return {'error': 'permission denied'}
+
+        # Launch task
+        self._delete_export()
+        # Make sure we nail down our state before sending off an asynchronous task.
+        self.save()
+        log.info("------------ in start_export - starting async task ---------------")
+        if my_api:
+            log.info("------------ in start_export - my_api found ---------------")
+        else:
+            log.info("------------ in start_export - my_api not found ---------------")
+
+        from .tasks import export_data as export_data_task  # Import here since this is edX LMS specific
+        async_result = export_data_task.delay(
+            # course_id not available in workbench.
+            unicode(getattr(self.runtime, 'course_id', 'course_id')),
+            root_block_id
+        )
+        if async_result.ready():
+            log.info("------------ in start_export- task ready ---------------")
+            log.info(async_result.id)
+            # In development mode, the task may have executed synchronously.
+            # Store the result now, because we won't be able to retrieve it later :-/
+            if async_result.successful():
+                # Make sure the result can be represented as JSON, since the non-eager celery
+                # requires that
+                json.dumps(async_result.result)
+            self._save_result(async_result)
```

Comment:
This logic is subtly different from what exists in `self._check_pending_export()`.  Would it be better to save the async_result.id to `self.active_export_task_id` first, and then call that method, so that saving always happens the same way?